### PR TITLE
EN-1981-Fix-checkbox-round-tripping-in-PAM-forms

### PR DESCRIPTION
**Pull request by alepintus**
Hi @joewhite101 ,
this is the cherry-pick from [this PR](https://github.com/entando/entando-components/pull/412)

Thanks!

### CODE REVIEW

**Comment by entando-service** [comment]
All is Well

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/system/services/kie/helper/FormToBpmHelper.java [hunk 1]
```diff
@@ -142,12 +144,17 @@ private static JSONObject buildSections(Map<String, JSONObject> jmap,
                 //Boolean checkbox values need to get round tripped as false. Without it data erasure can occur
                 //when a box is un-checked by a user. The input data from the user won't contain a key since the value
                 //isn't sent by the FE
-                if (formField.getType().equals("CheckBox") && !input.containsKey(parameter)) {
+                if ((formField.getType().equals("CheckBox") && !input.containsKey(parameter))
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/system/services/kie/helper/FormToBpmHelper.java]: Does this have test coverage

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/system/services/kie/helper/FormToBpmHelper.java [hunk 2]
```diff
@@ -112,28 +111,31 @@ private static JSONObject buildSections(Map<String, JSONObject> jmap,
...
             String parameter = formField.getName();
             String[] tok = parameter.split(SEPARATOR);
 
             if (tok.length > 1) {
                 String section = tok[0];
                 String field = tok[1];
+                //Boolean checkbox values need to get round tripped as false. Without it data erasure can occur
+                //when a box is un-checked by a user. The input data from the user won't contain a key since the value
+                //isn't sent by the FE
+                if (formField.getType().equals("CheckBox") && !input.containsKey(parameter)) {
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/system/services/kie/helper/FormToBpmHelper.java]: Needs some test coverage here. Even though a small change this is a key output so it needs to be tested

**Comment by entando-service** [comment]
All is Well

**Pull request merged**